kratos/python_scripts/process_factory.py

```diff
@@ -13,33 +13,64 @@ def ConstructListOfProcesses( self, process_list ):
         constructed_processes = []
         for i in range(0,process_list.size()):
             item = process_list[i]
-            if not item.Has("python_module"):
-                KM.Logger.PrintWarning("Your list of processes: ", process_list)
-                raise NameError('"python_module" must be defined in your parameters. Check all your processes')
 
-            # python-script that contains the process
-            python_module_name = item["python_module"].GetString()
+            # Registry-based instantiation
+            if item.Has("name"):
+                registry_entry = item["name"].GetString()
+                if KM.Registry.HasItem(registry_entry):
+                    # Get already stored prototype
+                    if KM.Registry.HasItem(f"{registry_entry}.Prototype"):
+                        prototype = KM.Registry[f"{registry_entry}.Prototype"]
+                    # Get prototype from stored Python module
+                    elif KM.Registry.HasItem(f"{registry_entry}.ModuleName"):
+                        class_name = registry_entry.split(".")[-1]
+                        module_name = KM.Registry[f"{registry_entry}.ModuleName"]
+                        module = import_module(module_name)
```

Comment:
shall we check before import?

Reply:
I'm not getting your point at all...

If you mean checking that current item has `name` (i.e. is to be constructed from the registry) it is done in line 18. If you mean checking that provided `name` is registered, it is done in line 20.

Alternatively, if you mean a try-except in the module import, I think the `import_module` handles this.

Reply:
I am saying if we should check if the module is already imported. Indeed, very small detail....